File: app/src/main/java/soup/movie/di/ui/TheaterEditUiModule.java
package soup.movie.di.ui;

import dagger.Module;
import dagger.Provides;
import soup.movie.data.MoobRepository;
import soup.movie.di.scope.ActivityScope;
import soup.movie.settings.TheaterSetting;
import soup.movie.ui.theater.edit.TheaterEditContract;
import soup.movie.ui.theater.edit.TheaterEditPresenter;

@Module
public class TheaterEditUiModule {

    @ActivityScope
    @Provides
    TheaterEditContract.Presenter presenter(
            MoobRepository moobRepository,
            TheaterSetting theaterSetting) {
        return new TheaterEditPresenter(moobRepository, theaterSetting);
    }
}

File: app/src/main/java/soup/movie/di/ui/TheaterEditUiModule.kt
package soup.movie.di.ui

import dagger.Module
import dagger.android.ContributesAndroidInjector
import soup.movie.di.scope.FragmentScope
import soup.movie.theater.di.TheaterAssistedInjectModule
import soup.movie.theater.di.TheaterEditDomainModule
import soup.movie.theater.di.TheaterEditFragmentModule
import soup.movie.theater.di.TheaterSortFragmentModule
import soup.movie.theater.edit.TheaterEditFragment
import soup.movie.theater.sort.TheaterSortFragment

@Module
interface TheaterEditUiModule {

    @FragmentScope
    @ContributesAndroidInjector(
        modules = [
            TheaterSortFragmentModule::class,
            TheaterAssistedInjectModule::class
        ]
    )
    fun bindTheaterSortFragment(): TheaterSortFragment

    @FragmentScope
    @ContributesAndroidInjector(
        modules = [
            TheaterEditFragmentModule::class,
            TheaterEditDomainModule::class,
            TheaterAssistedInjectModule::class
        ]
    )
    fun bindTheaterEditFragment(): TheaterEditFragment
}
